### PR TITLE
CVSL-1482 Improvements to CVL API documentation

### PR DESCRIPTION
**Pull request by gurps-singh**
This PR is to improve some of the Swagger documentation and naming conventions used for the CVL Public API. 

Screenshot 2023-11-20 at 15 50 05

Screenshot 2023-11-20 at 15 50 30

Screenshot 2023-11-20 at 15 50 42

Screenshot 2023-11-20 at 15 50 49

Screenshot 2023-11-20 at 15 50 56

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/config/OpenApiConfiguration.kt [hunk 1]
```diff
@@ -12,6 +12,22 @@ import org.springframework.context.annotation.Configuration
...
+      API for access to licence data.
+      
+      Please note:
+      - Variations and edits to licences are classed as new licences
+      - In flight licences are licences with the following status:
+        - [ACTIVE, IN_PROGRESS, SUBMITTED, APPROVED, VARIATION_IN_PROGRESS, VARIATION_SUBMITTED, VARIATION_APPROVED &
+      VARIATION_REJECTED]
+      - Condition data is different dependent on the type of condition: 
+          - Standard conditions have static text
+          - Additional conditions have a combination of template and inputted text
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/config/OpenApiConfiguration.kt:26]: "static and inputted text"?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/model/licence/Licence.kt [hunk 2]
```diff
@@ -13,7 +13,11 @@ data class Licence(
   @Schema(description = "The licence type", example = "AP")
   val licenceType: LicenceType,
 
-  @Schema(description = "The policy version number the licence is currently on", example = "2.1")
+  @Schema(
+    description = "The policy version number the licence is currently on",
+    example = "2.1",
+    allowableValues = ["1.0", "2.0", "2.1"],
+  )
   val policyVersion: String,
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/model/licence/Licence.kt:21]: mmh.. we could possibly move from a string over to an enum instance..

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/model/licence/Licence.kt [hunk 3]
```diff
@@ -22,36 +26,43 @@ data class Licence(
...
   val version: String,
 
-  @Schema(description = "The current status code for the licence", example = "IN_PROGRESS")
+  @Schema(
+    description = "The current status code for the licence. This is non exhaustive and subject to change",
+    example = "IN_PROGRESS",
+    allowableValues = [
+      "IN_PROGRESS", "SUBMITTED", "APPROVED", "ACTIVE", "VARIATION_IN_PROGRESS",
+      "VARIATION_SUBMITTED", "VARIATION_APPROVED",
+    ],
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/model/licence/Licence.kt:35]: I think this is prepopulated by the LicenceStatus enum? Probably better not to have to keep this list up to date?

Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/model/licence/Licence.kt:35]: the benefit of having our own licence status enum is that we can fix it to only have the allowed values

Comment (gurps-singh) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/model/licence/Licence.kt:35]: The current LicenceStatus enum contains statuses which we discussed might not be needed hence this version of it (which is missing VARIATION_REJECTED). Note below from [here](https://docs.google.com/document/d/1CrlAGx19jT43Uia1a_0CWPz9BG-CnkaoX4VGn5LRHC4/edit):

- statusCode - this is non exhaustive and subject to change
- rejected does not exist but VARIATION_REJECTED does
- remove INACTIVE

Comment (gurps-singh) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/model/licence/Licence.kt:35]: Ignore me - just found the Public API specific LicenceStatus enum

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/model/licence/Licence.kt (hunk 2 above)
Comment (gurps-singh) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/model/licence/Licence.kt]: Moved to enum instance

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request closed without merging**